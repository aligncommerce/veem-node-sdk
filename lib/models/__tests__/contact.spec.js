import ContactModel from '../contact'
import merge from 'lodash/merge'
import CONTACT_MOCK from './__mocks__/contact'

describe('Contact Model', () => {
  it('should match the snapshot', () => {
    expect(ContactModel.schema).toMatchSnapshot()
  })

  it('should filter out invalid properties', () => {
    const invalidContactRequest = merge({}, CONTACT_MOCK, {
      foo: 'bar',
      amount: {
        foo: 'bar',
      },
      payee: {
        foo: 'bar',
      },
    })

    const contact = new ContactModel(invalidContactRequest)
    const actualContactRequest = contact.request()

    expect(actualContactRequest).to.deep.equal(CONTACT_MOCK)
  })
})
