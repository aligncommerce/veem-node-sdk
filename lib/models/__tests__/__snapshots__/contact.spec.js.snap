// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Model request should match the snapshot 1`] = `
Object {
  "properties": Object {
    "batchItemId": Object {
      "type": "number",
    },
    "businessName": Object {
      "type": "string",
    },
    "contactAccountId": Object {
      "type": "number",
    },
    "dialCode": Object {
      "type": "string",
    },
    "email": Object {
      "type": "string",
    },
    "firstName": Object {
      "type": "string",
    },
    "id": Object {
      "type": "number",
    },
    "isoCountryCode": Object {
      "type": "string",
    },
    "lastName": Object {
      "type": "string",
    },
    "phoneNumber": Object {
      "type": "string",
    },
  },
  "title": "ContactRequest",
  "type": "object",
}
`;

exports[`Contact Model response should match the snapshot 1`] = `
Object {
  "properties": Object {
    "bankAccount": Object {
      "properties": Object {
        "bankAccountNumber": Object {
          "type": "string",
        },
        "bankAddress": Object {
          "properties": Object {
            "city": Object {
              "type": "string",
            },
            "line1": Object {
              "type": "string",
            },
            "line2": Object {
              "type": "string",
            },
            "postalCode": Object {
              "type": "string",
            },
            "stateProvince": Object {
              "type": "string",
            },
          },
          "type": "object",
        },
        "bankCnaps": Object {
          "type": "string",
        },
        "bankCode": Object {
          "type": "string",
        },
        "bankIfscBranchCode": Object {
          "type": "string",
        },
        "bankInstitutionNumber": Object {
          "type": "string",
        },
        "bankName": Object {
          "type": "string",
        },
        "beneficiaryName": Object {
          "type": "string",
        },
        "branchCode": Object {
          "type": "string",
        },
        "bsbBankCode": Object {
          "type": "string",
        },
        "clabe": Object {
          "type": "string",
        },
        "currencyCode": Object {
          "type": "string",
        },
        "iban": Object {
          "type": "string",
        },
        "isoCountryCode": Object {
          "type": "string",
        },
        "routingNumber": Object {
          "type": "string",
        },
        "sortCode": Object {
          "type": "string",
        },
        "swiftBic": Object {
          "type": "string",
        },
        "transitCode": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "businessAddress": Object {
      "properties": Object {
        "city": Object {
          "type": "string",
        },
        "line1": Object {
          "type": "string",
        },
        "line2": Object {
          "type": "string",
        },
        "postalCode": Object {
          "type": "string",
        },
        "stateProvince": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "businessName": Object {
      "type": "string",
    },
    "email": Object {
      "type": "string",
    },
    "externalBusinessId": Object {
      "type": "string",
    },
    "firstName": Object {
      "type": "string",
    },
    "isoCountryCode": Object {
      "type": "string",
    },
    "lastName": Object {
      "type": "string",
    },
    "phoneDialCode": Object {
      "type": "string",
    },
    "phoneNumber": Object {
      "type": "string",
    },
    "type": Object {
      "type": "string",
    },
  },
  "title": "ContactResponse",
  "type": "object",
}
`;
